pubsub/envelope_test.go

```diff
@@ -103,63 +103,75 @@ func TestCreateFromJSON(t *testing.T) {
 func TestCreateCloudEventsEnvelopeDefaults(t *testing.T) {
 	t.Run("default event type", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
-			"mypubsub", "", nil, "", "")
+			"mypubsub", "", nil, "", "", "", "")
 		assert.Equal(t, DefaultCloudEventType, envelope[TypeField])
 	})
 
 	t.Run("non-default event type", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "e1", "", "",
-			"mypubsub", "", nil, "", "")
+			"mypubsub", "", nil, "", "", "", "")
 		assert.Equal(t, "e1", envelope[TypeField])
 	})
 
 	t.Run("spec version", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
-			"mypubsub", "", nil, "", "")
+			"mypubsub", "", nil, "", "", "", "")
 		assert.Equal(t, CloudEventsSpecVersion, envelope[SpecVersionField])
 	})
 
 	t.Run("quoted data", func(t *testing.T) {
 		list := []string{"v1", "v2", "v3"}
 		data := strings.Join(list, ",")
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
-			"mypubsub", "", []byte(data), "", "")
+			"mypubsub", "", []byte(data), "", "", "", "")
 		t.Logf("data: %v", envelope[DataField])
 		assert.Equal(t, "text/plain", envelope[DataContentTypeField])
 		assert.Equal(t, data, envelope[DataField].(string))
 	})
 
 	t.Run("has subject", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "subject1", "",
-			"mypubsub", "", []byte(""), "", "")
+			"mypubsub", "", []byte(""), "", "", "", "")
 		t.Logf("data: %v", envelope[DataField])
 		assert.Equal(t, "subject1", envelope[SubjectField])
 	})
 
 	t.Run("no subject", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "topic",
-			"mypubsub", "", []byte(""), "", "")
+			"mypubsub", "", []byte(""), "", "", "", "")
 		t.Logf("data: %v", envelope[DataField])
 		assert.Empty(t, envelope[SubjectField])
 	})
 
 	t.Run("string data content type", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
-			"mypubsub", "", []byte("data"), "", "")
+			"mypubsub", "", []byte("data"), "", "", "", "")
 		assert.Equal(t, "text/plain", envelope[DataContentTypeField])
 	})
 
 	t.Run("trace id", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
-			"mypubsub", "", []byte("data"), "1", "")
+			"mypubsub", "", []byte("data"), "1", "", "", "")
 		assert.Equal(t, "1", envelope[TraceParentField])
 	})
 
 	t.Run("trace state", func(t *testing.T) {
 		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
-			"mypubsub", "", []byte("data"), "1", "key=value")
+			"mypubsub", "", []byte("data"), "1", "key=value", "", "")
 		assert.Equal(t, "key=value", envelope[TraceStateField])
 	})
+
+	t.Run("actor type", func(t *testing.T) {
+		envelope := NewCloudEventsEnvelope("a", "source", "", "", "",
+			"mypubsub", "", []byte("data"), "1", "", "actortype", "")
```

Comment:
Can you make this something like "myactortype" and then check for that value below? Tiny change :)